Automattic/jetpack
# User invitation: Mail the newly added user with WPCom invite

Fixes https://github.com/Automattic/wp-calypso/issues/86737 and https://github.com/Automattic/wp-calypso/issues/86738
This is part of the pdDR7T-1ja-p2

## Proposed changes:


Currently, when adding a new user to a site via wp-admin, it only creates the user in the site context (locally) therefore these users a not mapped as WPCom users.
This PR:
- improves it by intercepting the core email functionality and sending an invitation to the user via email. Once accepted, the user will not only be a local user but a WPCom user.
- It informs the site owner that the invite will be sent.

## Testing instructions:






- Pull this branch
- Sync the changes with your atomic site sync `jetpack rsync`
- Navigate to your atomic site {YOUR_SITE}/wp-admin/user-new.php
- You should see this message informing about the email invitation
 <img width="1066" alt="image" src="https://github.com/Automattic/jetpack/assets/2653810/92872e34-0d2b-477c-9cdc-9e74a5dab37d">
- When adding a new user, the user should get an email invite
 <img width="779" alt="image" src="https://github.com/Automattic/jetpack/assets/2653810/42b56cab-0c19-4d2e-8ed9-1a1dbec1966d">




### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?
